js/stores/message_store.js

```diff
@@ -22,18 +24,39 @@ var MessageStore = assign({}, EventEmitter.prototype, {
   all: function() {
     return _messages;
   },
+
+  local: function() {
+    var localMessages = [];
+    _messages.forEach(function(message) {
+      if (message.local === true) {
+        localMessages.push(message);
+      }
+    });
+
+    return localMessages;
+  }
 });
 
 MessageStore.dispatchToken = AppDispatcher.register(function(action) {
-  if (action.type === ActionTypes.NEW_MESSAGE) {
-    var message = {
-      id:  MessageStore.all().length,
-      timestamp: action.data.timestamp,
-      user: action.data.user,
-      text: action.data.text
-    };
-    _messages.push(message);
-    MessageStore.emit('change');
+  switch(action.type) {
+    case ActionTypes.NEW_MESSAGE:
+      var message = {
+        id:  MessageStore.all().length,
+        timestamp: action.data.timestamp,
+        user: action.data.user,
+        text: action.data.text,
+        local: action.data.local
+      };
+      _messages.push(message);
+      MessageStore.emit('change');
+      break;
+
+    case ActionTypes.EDIT_LAST_MESSAGE:
+      var localMessages = MessageStore.local();
+      var lastMessage = localMessages[localMessages.length - 1];
+      lastMessage.text = lastMessage.text.replace(action.data.find, action.data.replaceWith);
+      MessageStore.emit('change');
+      break;
   }
```

Comment:
as an alternative to switch/case, i've also seen it where it's like an object with action names and functions:
```
var handlers = {
  NEW_MESSAGE: function() {
  }
};
handlers[NEW_MESSAGE]();
```
something like that.. though i see that `ActionTypes.NEW_MESSAGE` is actually a pointer and can't be used as a key in itself so maybe not.. whatever you feel is cleaner :stuck_out_tongue: